### ISSUE TITLE
Build errors should indicate file name instead of path

### ISSUE THREAD

**Issue by Y0hy0h** [dillonkearns/elm-pages#103]
When a page has an error (metadata or document parsing), the error output mentions the URL path. This is confusing when an error occurrs in the root `index` file, because the path is empty.

Currently we get:
```
ERROR in -- METADATA DECODE ERROR ----------------------------------------------------- elm-pages

I ran into a problem when parsing the metadata for the page with this path:

Problem with the given value:

{
        "title": "elm-pages-starter - a simple blog starter",
        "typxe": "page"
    }

Expecting an OBJECT with a field named `type`
```

It would be helpful to either represent the index explicitely (e.g., with a slash as `/`) or print the actual file name. Printing the file name is nice, because that's where the fix has to occurr anyway.
```
ERROR in -- METADATA DECODE ERROR ----------------------------------------------------- elm-pages

I ran into a problem when parsing the metadata for the page from this file: index.md

Problem with the given value:

{
        "title": "elm-pages-starter - a simple blog starter",
        "typxe": "page"
    }

Expecting an OBJECT with a field named `type`
```

You can reproduce this in the [elm-pages-starter](https://github.com/dillonkearns/elm-pages-starter) by breaking the metadata of the `content/index.md` (add an `x` to `type`) and running `npm run build` or `npm run start`.

**Comment by Y0hy0h** [comment]
Addressed in #132.

**Comment by dillonkearns** [comment]
Longer term, it would be cool to show the exact source of the error. The design for this will change a lot with programmatic static routes, so might be worth revisiting the idea after that.
